Compilation fails on Python 3.11

With Python 3.11 (from CPython 9e05da62241):

```
  ERROR: Failed building wheel for typed-ast
  Running setup.py clean for typed-ast
Failed to build typed-ast
Installing collected packages: typing-extensions, toml, mypy-extensions, types-ujson, types-six, types-setuptools, types-requests, types-PyYAML, types-atomicwrites, typed-ast, mypy
    Running setup.py install for typed-ast: started
    Running setup.py install for typed-ast: finished with status 'error'
    ERROR: Command errored out with exit status 1:
     command: /mnt/home-extra/gsnedders/Documents/other-projects/wpt/web-platform-tests/tools/.tox/py311-mypy/bin/python -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-n2i2a0we/typed-ast_17123afaf5934a17827f262904008b6d/setup.py'"'"'; __file__='"'"'/tmp/pip-install-n2i2a0we/typed-ast_17123afaf5934a17827f262904008b6d/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-oz_exif7/install-record.txt --single-version-externally-managed --compile --install-headers /mnt/home-extra/gsnedders/Documents/other-projects/wpt/web-platform-tests/tools/.tox/py311-mypy/include/site/python3.11/typed-ast
         cwd: /tmp/pip-install-n2i2a0we/typed-ast_17123afaf5934a17827f262904008b6d/
    Complete output (63 lines):
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.11
    creating build/lib.linux-x86_64-3.11/typed_ast
    copying typed_ast/conversions.py -> build/lib.linux-x86_64-3.11/typed_ast
    copying typed_ast/ast3.py -> build/lib.linux-x86_64-3.11/typed_ast
    copying typed_ast/ast27.py -> build/lib.linux-x86_64-3.11/typed_ast
    copying typed_ast/__init__.py -> build/lib.linux-x86_64-3.11/typed_ast
    package init file 'ast3/tests/__init__.py' not found (or not a regular file)
    creating build/lib.linux-x86_64-3.11/typed_ast/tests
    copying ast3/tests/test_basics.py -> build/lib.linux-x86_64-3.11/typed_ast/tests
    running build_ext
    building '_ast27' extension
    creating build/temp.linux-x86_64-3.11
    creating build/temp.linux-x86_64-3.11/ast27
    creating build/temp.linux-x86_64-3.11/ast27/Custom
    creating build/temp.linux-x86_64-3.11/ast27/Parser
    creating build/temp.linux-x86_64-3.11/ast27/Python
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -march=x86-64 -mtune=native -O3 -pipe -fno-plt -fno-semantic-interposition -march=x86-64 -mtune=native -O3 -pipe -fno-plt -march=x86-64 -mtune=native -O3 -pipe -fno-plt -fPIC -Iast27/Include -I/mnt/home-extra/gsnedders/Documents/other-projects/wpt/web-platform-tests/tools/.tox/py311-mypy/include -I/usr/include/python3.11 -c ast27/Custom/typed_ast.c -o build/temp.linux-x86_64-3.11/ast27/Custom/typed_ast.o
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -march=x86-64 -mtune=native -O3 -pipe -fno-plt -fno-semantic-interposition -march=x86-64 -mtune=native -O3 -pipe -fno-plt -march=x86-64 -mtune=native -O3 -pipe -fno-plt -fPIC -Iast27/Include -I/mnt/home-extra/gsnedders/Documents/other-projects/wpt/web-platform-tests/tools/.tox/py311-mypy/include -I/usr/include/python3.11 -c ast27/Parser/acceler.c -o build/temp.linux-x86_64-3.11/ast27/Parser/acceler.o
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -march=x86-64 -mtune=native -O3 -pipe -fno-plt -fno-semantic-interposition -march=x86-64 -mtune=native -O3 -pipe -fno-plt -march=x86-64 -mtune=native -O3 -pipe -fno-plt -fPIC -Iast27/Include -I/mnt/home-extra/gsnedders/Documents/other-projects/wpt/web-platform-tests/tools/.tox/py311-mypy/include -I/usr/include/python3.11 -c ast27/Parser/bitset.c -o build/temp.linux-x86_64-3.11/ast27/Parser/bitset.o
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -march=x86-64 -mtune=native -O3 -pipe -fno-plt -fno-semantic-interposition -march=x86-64 -mtune=native -O3 -pipe -fno-plt -march=x86-64 -mtune=native -O3 -pipe -fno-plt -fPIC -Iast27/Include -I/mnt/home-extra/gsnedders/Documents/other-projects/wpt/web-platform-tests/tools/.tox/py311-mypy/include -I/usr/include/python3.11 -c ast27/Parser/grammar.c -o build/temp.linux-x86_64-3.11/ast27/Parser/grammar.o
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -march=x86-64 -mtune=native -O3 -pipe -fno-plt -fno-semantic-interposition -march=x86-64 -mtune=native -O3 -pipe -fno-plt -march=x86-64 -mtune=native -O3 -pipe -fno-plt -fPIC -Iast27/Include -I/mnt/home-extra/gsnedders/Documents/other-projects/wpt/web-platform-tests/tools/.tox/py311-mypy/include -I/usr/include/python3.11 -c ast27/Parser/grammar1.c -o build/temp.linux-x86_64-3.11/ast27/Parser/grammar1.o
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -march=x86-64 -mtune=native -O3 -pipe -fno-plt -fno-semantic-interposition -march=x86-64 -mtune=native -O3 -pipe -fno-plt -march=x86-64 -mtune=native -O3 -pipe -fno-plt -fPIC -Iast27/Include -I/mnt/home-extra/gsnedders/Documents/other-projects/wpt/web-platform-tests/tools/.tox/py311-mypy/include -I/usr/include/python3.11 -c ast27/Parser/node.c -o build/temp.linux-x86_64-3.11/ast27/Parser/node.o
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -march=x86-64 -mtune=native -O3 -pipe -fno-plt -fno-semantic-interposition -march=x86-64 -mtune=native -O3 -pipe -fno-plt -march=x86-64 -mtune=native -O3 -pipe -fno-plt -fPIC -Iast27/Include -I/mnt/home-extra/gsnedders/Documents/other-projects/wpt/web-platform-tests/tools/.tox/py311-mypy/include -I/usr/include/python3.11 -c ast27/Parser/parser.c -o build/temp.linux-x86_64-3.11/ast27/Parser/parser.o
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -march=x86-64 -mtune=native -O3 -pipe -fno-plt -fno-semantic-interposition -march=x86-64 -mtune=native -O3 -pipe -fno-plt -march=x86-64 -mtune=native -O3 -pipe -fno-plt -fPIC -Iast27/Include -I/mnt/home-extra/gsnedders/Documents/other-projects/wpt/web-platform-tests/tools/.tox/py311-mypy/include -I/usr/include/python3.11 -c ast27/Parser/parsetok.c -o build/temp.linux-x86_64-3.11/ast27/Parser/parsetok.o
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -march=x86-64 -mtune=native -O3 -pipe -fno-plt -fno-semantic-interposition -march=x86-64 -mtune=native -O3 -pipe -fno-plt -march=x86-64 -mtune=native -O3 -pipe -fno-plt -fPIC -Iast27/Include -I/mnt/home-extra/gsnedders/Documents/other-projects/wpt/web-platform-tests/tools/.tox/py311-mypy/include -I/usr/include/python3.11 -c ast27/Parser/tokenizer.c -o build/temp.linux-x86_64-3.11/ast27/Parser/tokenizer.o
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -march=x86-64 -mtune=native -O3 -pipe -fno-plt -fno-semantic-interposition -march=x86-64 -mtune=native -O3 -pipe -fno-plt -march=x86-64 -mtune=native -O3 -pipe -fno-plt -fPIC -Iast27/Include -I/mnt/home-extra/gsnedders/Documents/other-projects/wpt/web-platform-tests/tools/.tox/py311-mypy/include -I/usr/include/python3.11 -c ast27/Python/Python-ast.c -o build/temp.linux-x86_64-3.11/ast27/Python/Python-ast.o
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -march=x86-64 -mtune=native -O3 -pipe -fno-plt -fno-semantic-interposition -march=x86-64 -mtune=native -O3 -pipe -fno-plt -march=x86-64 -mtune=native -O3 -pipe -fno-plt -fPIC -Iast27/Include -I/mnt/home-extra/gsnedders/Documents/other-projects/wpt/web-platform-tests/tools/.tox/py311-mypy/include -I/usr/include/python3.11 -c ast27/Python/asdl.c -o build/temp.linux-x86_64-3.11/ast27/Python/asdl.o
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -march=x86-64 -mtune=native -O3 -pipe -fno-plt -fno-semantic-interposition -march=x86-64 -mtune=native -O3 -pipe -fno-plt -march=x86-64 -mtune=native -O3 -pipe -fno-plt -fPIC -Iast27/Include -I/mnt/home-extra/gsnedders/Documents/other-projects/wpt/web-platform-tests/tools/.tox/py311-mypy/include -I/usr/include/python3.11 -c ast27/Python/ast.c -o build/temp.linux-x86_64-3.11/ast27/Python/ast.o
    ast27/Python/ast.c: In function ‘ast_for_import_stmt’:
    ast27/Python/ast.c:2781:13: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
     2781 |             if (!import_alias)
          |             ^~
    In file included from ast27/Python/../Include/Python-ast.h:3,
                     from ast27/Python/ast.c:7:
    ast27/Python/../Include/../Include/asdl.h:46:31: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘if’
       46 | #define asdl_seq_SET(S, I, V) (S)->elements[I] = (V)
          |                               ^
    ast27/Python/ast.c:2783:17: note: in expansion of macro ‘asdl_seq_SET’
     2783 |                 asdl_seq_SET(aliases, 0, import_alias);
          |                 ^~~~~~~~~~~~
    ast27/Python/ast.c:2788:17: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
     2788 |                 if (!import_alias)
          |                 ^~
    In file included from ast27/Python/../Include/Python-ast.h:3,
                     from ast27/Python/ast.c:7:
    ast27/Python/../Include/../Include/asdl.h:46:31: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘if’
       46 | #define asdl_seq_SET(S, I, V) (S)->elements[I] = (V)
          |                               ^
    ast27/Python/ast.c:2790:21: note: in expansion of macro ‘asdl_seq_SET’
     2790 |                     asdl_seq_SET(aliases, i / 2, import_alias);
          |                     ^~~~~~~~~~~~
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -march=x86-64 -mtune=native -O3 -pipe -fno-plt -fno-semantic-interposition -march=x86-64 -mtune=native -O3 -pipe -fno-plt -march=x86-64 -mtune=native -O3 -pipe -fno-plt -fPIC -Iast27/Include -I/mnt/home-extra/gsnedders/Documents/other-projects/wpt/web-platform-tests/tools/.tox/py311-mypy/include -I/usr/include/python3.11 -c ast27/Python/graminit.c -o build/temp.linux-x86_64-3.11/ast27/Python/graminit.o
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -march=x86-64 -mtune=native -O3 -pipe -fno-plt -fno-semantic-interposition -march=x86-64 -mtune=native -O3 -pipe -fno-plt -march=x86-64 -mtune=native -O3 -pipe -fno-plt -fPIC -Iast27/Include -I/mnt/home-extra/gsnedders/Documents/other-projects/wpt/web-platform-tests/tools/.tox/py311-mypy/include -I/usr/include/python3.11 -c ast27/Python/mystrtoul.c -o build/temp.linux-x86_64-3.11/ast27/Python/mystrtoul.o
    ast27/Python/mystrtoul.c: In function ‘Ta27OS_strtoul’:
    ast27/Python/mystrtoul.c:112:21: error: ‘_PyLong_DigitValue’ undeclared (first use in this function)
      112 |                 if (_PyLong_DigitValue[Py_CHARMASK(str[1])] >= 16) {
          |                     ^~~~~~~~~~~~~~~~~~
    ast27/Python/mystrtoul.c:112:21: note: each undeclared identifier is reported only once for each function it appears in
    error: command '/usr/bin/gcc' failed with exit code 1
```

See https://github.com/python/cpython/commit/5f09bb021a2862ba89c3ecb53e7e6e95a9e07e1d for this getting moved to an internal API

--- a/ast27/Python/mystrtoul.c
+++ b/ast27/Python/mystrtoul.c
@@ -78,6 +78,32 @@ static int digitlimit[] = {
 #error "Need table for SIZEOF_LONG"
 #endif
 
+/* Table of digit values for 8-bit string -> integer conversion.
+ * '0' maps to 0, ..., '9' maps to 9.
+ * 'a' and 'A' map to 10, ..., 'z' and 'Z' map to 35.
+ * All other indices map to 37.
+ * Note that when converting a base B string, a char c is a legitimate
+ * base B digit iff digitvalue[Py_CHARPyLong_MASK(c)] < B.
+ */
+unsigned char digitvalue[256] = {
+    37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37,
+    37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37,
+    37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37,
+    0,  1,  2,  3,  4,  5,  6,  7,  8,  9,  37, 37, 37, 37, 37, 37,
+    37, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24,
+    25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 37, 37, 37, 37, 37,
+    37, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24,
+    25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 37, 37, 37, 37, 37,
+    37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37,
+    37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37,
+    37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37,
+    37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37,
+    37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37,
+    37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37,
+    37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37,
+    37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37,
+};
+
 /*
 **      strtoul
 **              This is a general purpose routine for converting
@@ -109,7 +135,7 @@ Ta27OS_strtoul(register char *str, char **ptr, int base)
             ++str;
             if (*str == 'x' || *str == 'X') {
                 /* there must be at least one digit after 0x */
-                if (_PyLong_DigitValue[Py_CHARMASK(str[1])] >= 16) {
+                if (digitvalue[Py_CHARMASK(str[1])] >= 16) {
                     if (ptr)
                         *ptr = str;
                     return 0;
@@ -118,7 +144,7 @@ Ta27OS_strtoul(register char *str, char **ptr, int base)
                 base = 16;
             } else if (*str == 'o' || *str == 'O') {
                 /* there must be at least one digit after 0o */
-                if (_PyLong_DigitValue[Py_CHARMASK(str[1])] >= 8) {
+                if (digitvalue[Py_CHARMASK(str[1])] >= 8) {
                     if (ptr)
                         *ptr = str;
                     return 0;
@@ -127,7 +153,7 @@ Ta27OS_strtoul(register char *str, char **ptr, int base)
                 base = 8;
             } else if (*str == 'b' || *str == 'B') {
                 /* there must be at least one digit after 0b */
-                if (_PyLong_DigitValue[Py_CHARMASK(str[1])] >= 2) {
+                if (digitvalue[Py_CHARMASK(str[1])] >= 2) {
                     if (ptr)
                         *ptr = str;
                     return 0;
@@ -147,7 +173,7 @@ Ta27OS_strtoul(register char *str, char **ptr, int base)
             ++str;
             if (*str == 'b' || *str == 'B') {
                 /* there must be at least one digit after 0b */
-                if (_PyLong_DigitValue[Py_CHARMASK(str[1])] >= 2) {
+                if (digitvalue[Py_CHARMASK(str[1])] >= 2) {
                     if (ptr)
                         *ptr = str;
                     return 0;
@@ -162,7 +188,7 @@ Ta27OS_strtoul(register char *str, char **ptr, int base)
             ++str;
             if (*str == 'o' || *str == 'O') {
                 /* there must be at least one digit after 0o */
-                if (_PyLong_DigitValue[Py_CHARMASK(str[1])] >= 8) {
+                if (digitvalue[Py_CHARMASK(str[1])] >= 8) {
                     if (ptr)
                         *ptr = str;
                     return 0;
@@ -177,7 +203,7 @@ Ta27OS_strtoul(register char *str, char **ptr, int base)
             ++str;
             if (*str == 'x' || *str == 'X') {
                 /* there must be at least one digit after 0x */
-                if (_PyLong_DigitValue[Py_CHARMASK(str[1])] >= 16) {
+                if (digitvalue[Py_CHARMASK(str[1])] >= 16) {
                     if (ptr)
                         *ptr = str;
                     return 0;
@@ -203,7 +229,7 @@ Ta27OS_strtoul(register char *str, char **ptr, int base)
     ovlimit = digitlimit[base];
 
     /* do the conversion until non-digit character encountered */
-    while ((c = _PyLong_DigitValue[Py_CHARMASK(*str)]) < base) {
+    while ((c = digitvalue[Py_CHARMASK(*str)]) < base) {
         if (ovlimit > 0) /* no overflow check required */
             result = result * base + c;
         else { /* requires overflow check */
@@ -240,7 +266,7 @@ Ta27OS_strtoul(register char *str, char **ptr, int base)
 overflowed:
     if (ptr) {
         /* spool through remaining digit characters */
-        while (_PyLong_DigitValue[Py_CHARMASK(*str)] < base)
+        while (digitvalue[Py_CHARMASK(*str)] < base)
             ++str;
         *ptr = str;
     }

